### ISSUE TITLE
Property emit dozer connector `SnapshottingStart` and `SnapshottingDone` events

### ISSUE THREAD

**Issue by chubei** [getdozer/dozer#1888]
We need to think about what "snapshotting" means for dozer connector. Not sending `SnapshottingDone` is not an option because downstream is relying on this message to enable certain optimizations.

The solution in my mind is like following:

`LogOperation::SnapshottingDone` should not include `connection_name`, and is only emitted once all connection affecting this log is done snapshotting.

We can leave it for another PR.

_Originally posted by @chubei in https://github.com/getdozer/dozer/pull/1883#discussion_r1300828524_
